pkg/attacher/lister.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attacher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+
+	"google.golang.org/grpc"
+)
+
+// Lister implements list operations against a remote CSI driver.
+type VolumeLister interface {
+	// ListVolumes calls ListVolumes on the driver and returns a map with keys
+	// of VolumeID and values of the list of Node IDs that volume is published
+	// on
+	ListVolumes(ctx context.Context) (map[string][]string, error)
+}
+
+type volumeLister struct {
+	conn         *grpc.ClientConn
+	capabilities []csi.ControllerServiceCapability
+}
+
+var _ VolumeLister = &volumeLister{}
+
+// NewVolumeLister provides a new VolumeLister object.
+func NewVolumeLister(conn *grpc.ClientConn) VolumeLister {
+	return &volumeLister{
+		conn: conn,
+	}
+}
+
+func (a *volumeLister) ListVolumes(ctx context.Context) (map[string]([]string), error) {
+	client := csi.NewControllerClient(a.conn)
```

Comment:
Since conn is a field, maybe make client a field as well ?

Reply:
not sure I follow that logic.  It's a common pattern to save the connection and create a new client with it when necessary

Reply:
Alright